### ISSUE TITLE
Integer overflow not detected in math operators

### ISSUE THREAD

**Issue by PaulSnow** [DTRules/DTRules#91]
## Problem
Arithmetic operators (+, -, *) on integers do not check for overflow. Go will silently wrap around, causing incorrect calculations.

## Location
`go/pkg/dtrules/operators/math.go:86-128`

## Example
```go
// opAdd: (line 86)
return state.DataPush(dtrules.GetRIntegerValue(aVal + bVal))  // No overflow check
```

## Impact
Silent data corruption in calculations with large numbers.

## Suggested Fix
Add overflow detection and auto-promote to double:
```go
func opAdd(state dtrules.State) error {
    // ... get values ...
    if aVal > 0 && bVal > math.MaxInt64-aVal {
        // Overflow - promote to double
        return state.DataPush(dtrules.NewValueDouble(float64(aVal) + float64(bVal)))
    }
    return state.DataPush(dtrules.GetRIntegerValue(aVal + bVal))
}
```

## Priority
CRITICAL - Must fix before production use.

**Comment by PaulSnow** [comment]
Fixed: Added overflow detection to opAdd, opSub, opMul in operators/math.go. Auto-promotes to double when overflow would occur.
